### PR TITLE
Closes CPPL-59

### PR DESCRIPTION
**Pull request by iodream**
Create New list functionality

### CODE REVIEW

**Review by VasylDobrovolskyi — APPROVED** [review]

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 1]
```diff
@@ -27,11 +28,11 @@ void ListPagesController::ConnectListPage()
 		&ListsSubPage::GoToCreateList,
 		this,
 		&ListPagesController::OnGoToCreateList);
-	connect(
-		&m_list_page,
-		&ListsSubPage::GoToProducts,
-		this,
-		&ListPagesController::OnGoToProducts);
+//	connect(
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/ListPagesController.cpp:31]: Why did you do this?

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 2]
```diff
@@ -75,10 +82,42 @@ void ListPagesController::OnGoToCreateList()
...
 {
-	PageData data{};
-	data.setValue(list);
-	emit ChangeSubPage(MainSubPages::PRODUCTS, data);
-}
+	AddNewListsModel model{m_hostname};
+
+	List new_list;
+	new_list.name = m_create_page.GetListName();
+	new_list.id = m_list_page.get_list_size() + 1;
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/ListPagesController.cpp:92]: Set id to 0

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 3]
```diff
@@ -75,10 +82,42 @@ void ListPagesController::OnGoToCreateList()
 	emit ChangeSubPage(MainSubPages::CREATE_LIST);
 }
 
-void ListPagesController::OnGoToProducts(const List& list)
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/ListPagesController.cpp:78]: Restore this function

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 4]
```diff
@@ -75,10 +82,42 @@ void ListPagesController::OnGoToCreateList()
...
+	state.name = QString("");
+	new_list.state = state;
+
+	if(!model.CheckName(new_list.name))
+	{
+		emit Message(
+				QString("Error!"),
+				QString("List name can't be empty")
+				);
+		return; //abort
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/ListPagesController.cpp:106]: Useless comment

**Review by Denys-Bykov-0 — CHANGES_REQUESTED** [review]

**Review by MaksymTerekhovskyy — APPROVED** [review]

File: src/Client/Controller/MainPage/ListPagesController.cpp (hunk 1 above)
Comment (iodream) [reply src/Client/Controller/MainPage/ListPagesController.cpp:31]: sorry, i had undefined reference errors here. but i just retested and they're gone. function restored.

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 5]
```diff
@@ -75,10 +82,49 @@ void ListPagesController::OnGoToCreateList()
 	emit ChangeSubPage(MainSubPages::CREATE_LIST);
 }
 
+//checks data and sends request
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/ListPagesController.cpp:85]: Use comments only when that is hard to understand from the code itself. If you want to tell what function does it is better to do it in header, but in this case it rather redundant

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 6]
```diff
@@ -75,10 +82,49 @@ void ListPagesController::OnGoToCreateList()
...
+	{
+		emit Message(
+			QString("Error!"),
+			QString::fromStdString(response.reason));
+		return ;
+	}
+
+	auto lists = model.ParseResponse(response);
+
+	emit GoBack(); //immediately go to previous page
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/ListPagesController.cpp:122]: I can easily read this from the code

File: src/Client/Models/List/AddNewListModel.cpp [hunk 7]
```diff
@@ -10,6 +10,14 @@ Net::Request  AddNewListsModel::FormRequest(const List& list, const IdType& user
 	return request;
 }
 
+bool AddNewListsModel::CheckName(QString name)
+{
+	std::string str = name.toStdString();
+	str.erase(std::remove_if(str.begin(), str.end(), ::isspace)
```
Comment (Denys-Bykov-0) [inline src/Client/Models/List/AddNewListModel.cpp:16]: Weird format

**Review by Denys-Bykov-0 — APPROVED** [review]

**Pull request merged**